### PR TITLE
feat(mission_planner)!: support ad api

### PR DESCRIPTION
**Pull request by isamu-takagi**
Signed-off-by: Takagi, Isamu <isamu.takagi@tier4.jp>

## Description

Introduce AD API for routing. This PR is the mission planner part of the diagram.
Here are the changes for each package:
- component_interface_specs: Add the internal routing interface definitions.
- tier4_planning_launch: remove redundant launch file.
- mission_planner: change mission planner to support AD API.

planning-routing drawio

## Related links

https://github.com/autowarefoundation/autoware.universe/pull/1494
https://github.com/autowarefoundation/autoware_msgs/pull/33
https://github.com/autowarefoundation/autoware_adapi_msgs/pull/14

## Tests performed

1. Launch the planning simulation.
2. Check the routing state is UNSET.
3. Call the set_route_points API (refer to notes).
4. Check the routing state is SET.
5. Call the set_route_points API and check if it fails.
6. Start vehicle driving.
7. When vehicle goals, check the routing state is ARRIVED.
8. Call the clear_route API.
9. Call the set_route API (refer to notes)..

## Notes for reviewers

Sample Routes are from this position.
Screenshot from 2022-08-02 16-20-43

Sample1 (SetRoutePoints)
```bash
ros2 service call /planning/routing/set_route_points  autoware_ad_api_msgs/srv/SetRoutePoints "
header: {stamp: {sec: 1659424851, nanosec: 893215051}, frame_id: map}
goal:
  position: {x: 3835.50, y: 73765.91, z: 0.0}
  orientation: {x: 0.0, y: 0.0, z: -0.52530, w: 0.85091}"
```

Sample2 (SetRoutePoints)
```bash
ros2 service call /planning/routing/set_route_points  autoware_ad_api_msgs/srv/SetRoutePoints "
header: {stamp: {sec: 1659424851, nanosec: 893215051}, frame_id: map}
goal:
  position: {x: 3835.50, y: 73765.91, z: 0.0}
  orientation: {x: 0.0, y: 0.0, z: -0.52530, w: 0.85091}
waypoints:
 - position: {x: 3774.31, y: 73802.53, z: 0.0}
   orientation: {x: 0.0, y: 0.0, z: 0.22214, w: 0.97501}"
```

Sample1 (SetRoute)
```bash
ros2 service call /planning/routing/set_route  autoware_ad_api_msgs/srv/SetRoute "
header: {stamp: {sec: 1659424851, nanosec: 893215051}, frame_id: map}
goal:
  position: {x: 3835.50, y: 73765.91, z: 0.0}
  orientation: {x: 0.0, y: 0.0, z: -0.52530, w: 0.85091}
segments:
- {preferred: {id: 34976, type: lane}, alternatives: []}
- {preferred: {id: 34981, type: lane}, alternatives: []}
- {preferred: {id: 34585, type: lane}, alternatives: []}
- {preferred: {id: 34579, type: lane}, alternatives: []}
- {preferred: {id: 34780, type: lane}, alternatives: []}
- {preferred: {id: 34675, type: lane}, alternatives: []}"
```

Sample2 (SetRoute)
```bash
ros2 service call /planning/routing/set_route  autoware_ad_api_msgs/srv/SetRoute "
header: {stamp: {sec: 1659424851, nanosec: 893215051}, frame_id: map}
goal:
  position: {x: 3835.50, y: 73765.91, z: 0.0}
  orientation: {x: 0.0, y: 0.0, z: -0.52530, w: 0.85091}
segments:
- {preferred: {id: 34976, type: lane}, alternatives: []}
- {preferred: {id: 34981, type: lane}, alternatives: []}
- {preferred: {id: 34636, type: lane}, alternatives: []}
- {preferred: {id: 34564, type: lane}, alternatives: []}
- {preferred: {id: 34495, type: lane}, alternatives: []}
- {preferred: {id: 34507, type: lane}, alternatives: []}
- {preferred: {id: 34795, type: lane}, alternatives: []}
- {preferred: {id: 34606, type: lane}, alternatives: []}
- {preferred: {id: 34672, type: lane}, alternatives: []}
- {preferred: {id: 34675, type: lane}, alternatives: []}"
```

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by isamu-takagi** [comment]
@kenji-miyake @yukkysaito @mitsudome-r 
[In this PR](https://github.com/autowarefoundation/autoware.universe/pull/1559), we changed the root topic name to `/planning/mission_planning/route` (original name is `/planning/routing/route`). However, an error has occurred because the name is already used to distribute the HAD map route internally. So I'll change it to `/planning/mission_planning/api_route`. Or any other suggestions?

**Comment by kenji-miyake** [comment]
@isamu-takagi I was thinking that we'll replace the existing topic. Would it be possible for you to explain the problem a bit more concretely? :bow:

**Comment by isamu-takagi** [comment]
@kenji-miyake The existing topic is an implementation-dependent message type [HADMapRoute](https://github.com/tier4/autoware_auto_msgs/blob/tier4/main/autoware_auto_planning_msgs/msg/HADMapRoute.idl). The new topic is an implementation-independent API message type [Route](https://github.com/autowarefoundation/autoware.universe/blob/main/common/autoware_ad_api_msgs/routing/msg/Route.msg). So it cannot be replaced without forcing the internal implementation nodes to use the API type.

**Comment by isamu-takagi** [comment]
Waiting for https://github.com/autowarefoundation/autoware_msgs/pull/33.

**Comment by TakaHoribe** [comment]
When I launch the autoware with this PR merged, it failed with the following error:

```
[component_container-43] [ERROR 1665995713.014441264] [planning.mission_planning.mission_planning_container] OnLoadNode(): Failed to find class with the requested plugin name 'mission_planner::MissionPlannerLanelet2' in the loaded library:(L219)
```

**Comment by isamu-takagi** [comment]
@TakaHoribe 
Are you using TIER IV's meta-repository? Then, I'll sync it.

**Comment by TakaHoribe** [comment]
> Are you using TIER IV's meta-repository? I'll sync it.

Yes. Would you make a PR for the autoware_launch?

File: planning/mission_planner/src/mission_planner/arrival_checker.cpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "arrival_checker.hpp"
+
+#include <tier4_autoware_utils/tier4_autoware_utils.hpp>
+
+#include <tf2/utils.h>
+
+// TODO(Takagi, Isamu): remove when modified goal is always published
```
Comment (kosuke55) [inline planning/mission_planner/src/mission_planner/arrival_checker.cpp:21]: modified goal is always published with https://github.com/autowarefoundation/autoware.universe/pull/2094

**Comment by isamu-takagi** [comment]
> Yes. Would you make a PR for the autoware_launch?

@TakaHoribe https://github.com/tier4/autoware_launch/pull/520

**Review by TakaHoribe — APPROVED** [review]
Confirmed it works for existing scenarios.

**Pull request merged**